app/services/PaymentAPIService.scala

```diff
@@ -2,12 +2,28 @@ package services
 
 import java.io.IOException
 
+import io.circe.generic.JsonCodec
+import io.circe.parser.parse
+import codecs.CirceDecoders.decodeErrorWrapper
 import play.api.libs.json._
 import play.api.libs.ws.{WSClient, WSResponse}
 import services.ExecutePaymentBody._
+import monitoring.SafeLogger
 
 import scala.concurrent.{ExecutionContext, Future}
 
+class PaymentApiError extends Exception
+
+case class StripeApiError(exceptionType: Option[String], responseCode: Option[Int], requestId: Option[String], message: String) extends PaymentApiError() {
+  override val getMessage: String = message
+}
+
+case class PaypalApiError(responseCode: Option[Int], errorName: Option[String], message: String) extends PaymentApiError {
+  override val getMessage: String = message
+}
+
+case class ErrorWrapper(error: PaymentApiError, responseType: String)
```

Comment:
I don't think this class is necessary

Reply:
At the moment this is needed as the Payment exception sent from Payment-API is wrapped in an Error. I can remove this class and search the JSON for "error" using the \\ command, but this seemed a more readable way to handle it.

Reply:
I see; and completely agree this is a nicer way of handling it 👌. Keeping the 'wrapper' class, I would advocate for the following name changes:
- `PaymentApiError => PaymentProviderError`
- `ErrorWrapper => PaymentApiError`

Also, you can lose the `responseType` field - it's never used - in which case, you don't have to write a custom decoder using `Decoder.forProduct2`